pkg/lib/authn/otp/service.go

```diff
@@ -16,15 +16,26 @@ type CodeStore interface {
 	Delete(target string) error
 }
 
+type MagicLinkStore interface {
+	Create(token string, code *Code) error
```

Comment:
I think MagicLinkStore should only store the email but not the code model. The reason is to avoid confusion, the original code store should be the single source of truth.

Reply:
Updated